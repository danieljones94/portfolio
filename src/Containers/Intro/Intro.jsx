import React, { Component } from "react";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
// import Col from "react-bootstrap/Col";
import Column from "../../components/Column";
import styles from "./Intro.module.scss";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faChevronDown } from "@fortawesome/free-solid-svg-icons";

class Intro extends Component {
  state = { hover: false };

  // toggleHover = event => {
  //   this.setState({ hover: !this.state.hover });
  // };

  render() {
    // let style;
    // this.state.hover
    //   ? (style = { cursor: "pointer" })
    //   : (style = { transform: "scale(1.1)" });

    return (
      <>
        <section className={styles.introContainer} id="intro">
          <Container className={styles.gridContainer}>
            {/* Stack the columns on mobile by making one full-width and the other half-width */}
            <Row className={styles.row}>
              <Column
                className={styles.html}
                text={
                  "I have  experience working with HTML, CSS, SASS and Javascript, completing various challenges and personal projects"
                }
              />
              <Column
                className={styles.agile}
                text={
                  "I enjoy engaging in Agile and SCRUM principles, having personally ran and facilitated stand up meetings, retrospectives and kanban team boards."
                }
              />
            </Row>

            {/* Columns start at 50% wide on mobile and bump up to 33.3% wide on desktop */}
            <Row className={styles.row}>
              <Column
                className={styles.gitHub}
                text={
                  "Experience using GitHub flow and utilising version control during team based projects. Please see my GitHub for more info."
                }
              />
              <Column
                className={styles.database}
                text={
                  "An understanding of the use of databases. I have experience of storing, creating, updating and deleting data, as well as REST API practices."
                }
              />
              <Column
                className={styles.unitTesting}
                text={
                  "I understand the importance of unit testing and strive to use it where appropriate to make my code concise and readable."
                }
              />
            </Row>

            {/* Columns are always 50% wide, on mobile and desktop */}
            <Row className={styles.row}>
              <Column
                className={styles.react}
                text={
                  "Experience of using React.js building both personal and client projects. I have found a real enjoyment using React to build applications."
                }
              />
              <Column
                className={styles.softSkills}
                text={
                  "This is probably the skillset I value and pride myself on the most. Having a background in performance psychology, this is something I seek to continually improve."
                }
              />
            </Row>
          </Container>
        </section>
        <div>
          <a href="#work" className={styles.scroll}>
            <FontAwesomeIcon icon={faChevronDown} />
          </a>
        </div>
      </>
      //   <section className={`${styles.introContainer} ${style.}`}>
      //     <Panel title="HTML, SASS, JAVASCRIPT" description="Hello World" />
      //     <Panel title="React.js" description="Hello World" />
      //     <Panel title="OTHER TECH SKILLS" description="Hello Wolrd" />
      //     <Panel title="SOFT SKILLS" description="Hello World" />
      //   </section>
    );
  }
}

export default Intro;
