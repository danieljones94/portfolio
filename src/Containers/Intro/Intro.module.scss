.introContainer {
  height: 100vh;
  display: flex;
  justify-content: center;

  .gridContainer {
    margin-top: 5vh;
  }

  .row {
    height: 33.3%;
  }

  .html {
    background-image: url(../../images/photo-1454165205744-3b78555e5572.jfif);
    transition: all 0.5s ease-in-out;
  }

  .agile {
    background-image: url(../../images/photo-1512758017271-d7b84c2113f1.jfif);
    background-position: center;
    background-size: cover;
  }

  .database {
    background-image: url(../../images/DatabaseDesign_Infographic-1.png);
    background-position: center;
    background-size: contain;
    background-repeat: no-repeat;
  }

  .unitTesting {
    background-image: url(../../images/Unit-Testing.jpg);
    background-position: center;
    background-size: cover;
    background-repeat: no-repeat;
  }

  .gitHub {
    background-image: url(../../images/machine_learning_github.jpg);
    background-size: cover;
    background-position: center;
  }

  .react {
    background-image: url(../../images/1_y6C4nSvy2Woe0m7bWEn4BA.png);
    background-size: cover;
    background-position: center;
  }

  .softSkills {
    background-image: url(../../images/1A83FE99-F4E9-468B-BC04-1526DB47954C.JPEG);
    background-size: cover;
  }
}
