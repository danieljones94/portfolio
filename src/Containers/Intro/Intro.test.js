import React from "react";
import Intro from "./Intro";
import { shallow } from "enzyme";

let mockText =
  "A tug of war game, whereby the player who presses their button the most times wins.";

describe("Intro tests", () => {
  let component;
  beforeEach(() => {
    component = shallow(<Intro text={mockText} />);
  });

  test("should match the snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  test("there should be an a tag rendered", () => {
    expect(component.find("a").length).toEqual(1);
  });

  test("there should be test rendered out for each grid section", () => {
    expect(component.text()).toContain(mockText);
  });

  test("the hover state of the page should be false", () => {
    expect(component.state("hover").toEqual(false));
  });
});
