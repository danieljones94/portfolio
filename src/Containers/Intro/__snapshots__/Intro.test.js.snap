// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro tests should match the snapshot 1`] = `
<Fragment>
  <section
    className="introContainer"
    id="intro"
  >
    <Container
      className="gridContainer"
      fluid={false}
    >
      <ForwardRef
        className="row"
        noGutters={false}
      >
        <Column
          className="html"
          text="I have  experience working with HTML, CSS, SASS and Javascript, completing various challenges and personal projects"
        />
        <Column
          className="agile"
          text="I enjoy engaging in Agile and SCRUM principles, having personally ran and facilitated stand up meetings, retrospectives and kanban team boards."
        />
      </ForwardRef>
      <ForwardRef
        className="row"
        noGutters={false}
      >
        <Column
          className="gitHub"
          text="Experience using GitHub flow and utilising version control during team based projects. Please see my GitHub for more info."
        />
        <Column
          className="database"
          text="An understanding of the use of databases. I have experience of storing, creating, updating and deleting data, as well as REST API practices."
        />
        <Column
          className="unitTesting"
          text="I understand the importance of unit testing and strive to use it where appropriate to make my code concise and readable."
        />
      </ForwardRef>
      <ForwardRef
        className="row"
        noGutters={false}
      >
        <Column
          className="react"
          text="Experience of using React.js building both personal and client projects. I have found a real enjoyment using React to build applications."
        />
        <Column
          className="softSkills"
          text="This is probably the skillset I value and pride myself on the most. Having a background in performance psychology, this is something I seek to continually improve."
        />
      </ForwardRef>
    </Container>
  </section>
  <div>
    <a
      className="scroll"
      href="#work"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f078",
              "M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z",
            ],
            "iconName": "chevron-down",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </a>
  </div>
</Fragment>
`;
